### PR TITLE
Explicit modules

### PR DESCRIPTION
**Pull request by tresreid**
adds modules in explicit memory and relevant support in validation framework. 
Validation plots can be found here: https://www.classe.cornell.edu/~mgr85/SegmentLinking_keep/PR25/efficiencies/

### CODE REVIEW

**Review by sgnoohc — COMMENTED** [review]
Something is odd with the validation outputs.
https://www.classe.cornell.edu/~mgr85/SegmentLinking_keep/PR25/efficiencies/eff_comparison_plots__72e6987DIRTY_muonGun/plots.log
I think the issue here is that the validation benchmark file is assumed to exist but it does not.
Currently the benchmark results are stored in /nfs-7 area which I am guessing is not on cornell machine...
I did not foresee this... my bad.

**Comment by beiwang2003** [comment]
@tresreid I won't be able to check the code until next Tues. I have two 2-3 hours tutorials on Monday and will need to prepare that over the weekend. Is that ok?

File: bin/sdl.cc [hunk 1]
```diff
@@ -77,13 +77,13 @@ int main(int argc, char** argv)
 
     // A default value one
     if (ana.input_file_list_tstring.EqualTo("muonGun"))
-        ana.input_file_list_tstring = "/nfs-7/userdata/phchang/trackingNtuple/trackingNtuple_100_pt0p5_2p0.root";
+        ana.input_file_list_tstring = "trackingNtuple_100_pt0p5_2p0.root";
```
[the remark this answers is not in the document]
Comment (tresreid) [reply bin/sdl.cc:80]: Yeah I agree. I think we can make a path on both phi3 and lnx7188. I think something like "/data1/segmentlinking/..." or "/data2/...". We can store the ntuples (and cpu validation) there with the same path name.

**Comment by tresreid** [comment]
No problem and no rush. After taking to Philip I think that there are also a few things to still work out with explicit memory management when it comes to the physics validation.

File: bin/sdl.cc [hunk 2]
```diff
@@ -77,13 +77,13 @@ int main(int argc, char** argv)
 
     // A default value one
     if (ana.input_file_list_tstring.EqualTo("muonGun"))
-        ana.input_file_list_tstring = "/nfs-7/userdata/phchang/trackingNtuple/trackingNtuple_100_pt0p5_2p0.root";
+        ana.input_file_list_tstring = "/data2/segmentlinking/trackingNtuple_100_pt0p5_2p0.root";
```
Comment (slava77) [inline bin/sdl.cc:80]: on phi3 these files are in `/data2/segmentlinking/trackingNtuple/CMSSW_10_4_0_patch1-orig`

**Pull request merged**